build-image/action.yml

```diff
@@ -39,15 +45,36 @@ runs:
     - name: Read current git details
       id: current-git-details
       shell: bash
+      env:
+        SECURITY_PATCH: ${{ inputs.security_patch }}
       run: |
-        echo "short=sha-$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
-        echo "long=sha-$(git rev-parse HEAD)" >> $GITHUB_OUTPUT
+        short="sha-$(git rev-parse --short HEAD)"
+        long="sha-$(git rev-parse HEAD)"
+        tag=""
+        
+        # Add "security-patch" suffix, add random 4 characters to the end of the tag
+        if [ "$SECURITY_PATCH" == "true" ]; then
+          random_suffix=$(shuf -i 1000-9999 -n 1)
+          short="${short}-security-patch-${random_suffix}"
+          long="${long}-security-patch-${random_suffix}"
```

Comment:
In `check-image-exists`, we check for the existence of the short sha in ECR. Suffixing with `security-patch-${random_suffix}` helps create a unique tag that will force a new build